### ISSUE TITLE
ide-helper:generate is not working in production environment

### ISSUE THREAD

**Issue by lihaibh** [barryvdh/laravel-ide-helper#228]
I have an issue in my Laravel 5.1 project.
For my understanding and usage, this package should be used only on develop / acceptance environments to help my ide with auto complition and surely not on production environment, so ive added the dependency under "require-dev" scope and not under "require" scope  in composer.json file as given as an example in the documentation.

This action bring in some problems:
1. app.php should include service provider: "\Barryvdh\LaravelIdeHelper\IdeHelperServiceProvider" only on develop || acceptance environment, alright piece of cake, just added to my ApplicationServiceProvider a check if im not in production before register this service provider.

But there is another issue:
2. composer.json file has an attribute: "post-update-cmd" which tells composer to run commands after update command is executed.
so i want to run "php artisan ide-helper:generate" after composer update, but if im doing it on production environment i have an error ofcourse becuase the service provider of IdeHelper is not register on production environment.

Ive tried to replace:
on composer.json file under post-update-cmd attribute i tried to replace:
php artisan ide-helper:generate
With:
[[ $(php artisan env) == *develop[^a-zA-Z]* ]] && php artisan ide-helper:generate

So basically it should run the command only if the output of command 'php artisan env' contains chars "develop".
When i run it manually in the shell it works quite well but its not working with composer.

How do i tell composer to run specific commands when im exclusively run it on certain environments?

**Comment by gmsantosxl** [comment]
I don't think its safe to run `composer update` on production environment. Some dependency could break you code and will be a pain to debug it.

Instead when deploying to production I just run `composer install`, using the specific versions on `composer.lock` file and it does not fire `post-update-command`. Another option is using the [`--no-script ` option](https://getcomposer.org/doc/03-cli.md#update) on `composer update`.

**Comment by barryvdh** [comment]
Indeed, never run composer update on production!
